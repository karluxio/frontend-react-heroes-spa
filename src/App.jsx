function App() {

  return (
    <>
      <h1>Heroes App 🦸🏼‍♂️</h1>
      <hr />
    </>
  )
}

export default App
