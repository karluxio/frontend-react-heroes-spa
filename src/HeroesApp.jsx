function HeroesApp() {

  return (
    <>
      <h1>Heroes App 🦸🏼‍♂️</h1>
      <hr />
    </>
  )
}

export default HeroesApp
