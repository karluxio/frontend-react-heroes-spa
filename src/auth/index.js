export * from './context';
export * from './context/types';
export * from './pages/LoginPage';
