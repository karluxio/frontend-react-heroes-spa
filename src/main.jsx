import React from 'react'
import ReactDOM from 'react-dom/client'

import HeroesApp from './HeroesApp'

import './styles.css'
import { BrowserRouter } from '../node_modules/react-router-dom/dist/index'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <BrowserRouter>
      <HeroesApp />
    </BrowserRouter>
  </React.StrictMode>,
)
