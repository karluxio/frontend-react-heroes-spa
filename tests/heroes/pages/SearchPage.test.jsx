import { render, screen, fireEvent } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'

import { SearchPage } from '../../../src/heroes'

const mockedUseNavigate = jest.fn()

jest.mock("react-router-dom", () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUseNavigate
}))

describe('SearchPage', () => {

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it("should render with default values", () => {
    const { container } = render(
      <MemoryRouter>
        <SearchPage />
      </MemoryRouter>
    )

    // screen.debug()
    expect(container).toMatchSnapshot()
  })

  it("should render batman and the input value like query string", () => {
    render(
      <MemoryRouter initialEntries={['/search?q=batman']}>
        <SearchPage />
      </MemoryRouter>
    )

    const input = screen.getByRole('textbox')
    expect(input.value).toBe("batman")

    const img = screen.getByRole('img')
    expect(img.src).toContain('dc-batman.jpg')

    const divSearch = screen.getByLabelText('search-alert')
    expect(divSearch.innerHTML).toContain('Search a hero')

    const divError = screen.getByLabelText('error-alert')
    expect(divError.style.display).toContain('none')
  })

  it("should render an error if hero not exists", () => {
    render(
      <MemoryRouter initialEntries={['/search?q=rino']}>
        <SearchPage />
      </MemoryRouter>
    )

    const divAlert = screen.getByLabelText('search-alert')
    expect(divAlert.style.display).toContain('none')

    const divError = screen.getByLabelText('error-alert')
    expect(divError.style.display).toContain('')
  })

  it("should call navigate to move new screen", () => {

    const inputValue = 'superman'

    render(
      <MemoryRouter initialEntries={['/search']}>
        <SearchPage />
      </MemoryRouter>
    )

    const input = screen.getByRole('textbox')
    fireEvent.change(input, { target: { name: 'searchText', value: `${inputValue}` } })
    fireEvent.submit(screen.getByLabelText('form'))

    expect(mockedUseNavigate).toHaveBeenCalled()
    expect(mockedUseNavigate).toHaveBeenCalledWith(`/search?q=${inputValue}`)
  })
})