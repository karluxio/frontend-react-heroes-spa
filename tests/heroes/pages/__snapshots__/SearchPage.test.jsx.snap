// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage should render with default values 1`] = `
<div>
  <div
    class="animate__animated animate__fadeIn"
  >
    <h1>
      Search
    </h1>
    <hr />
    <div
      class="row"
    >
      <div
        class="col-5"
      >
        <h4>
          Searching...
        </h4>
        <hr />
        <form
          aria-label="form"
        >
          <input
            autocomplete="off"
            class="form-control"
            name="searchText"
            placeholder="search a hero like spiderman..."
            type="text"
            value=""
          />
          <button
            class="btn btn-outline-primary mt-2"
          >
            Search
          </button>
        </form>
      </div>
      <div
        class="col-7"
      >
        <h4>
          Results
        </h4>
        <hr />
        <div
          aria-label="search-alert"
          class="alert alert-primary animate__animated animate__fadeIn"
        >
          Search a hero
        </div>
        <div
          aria-label="error-alert"
          class="alert alert-danger animate__animated animate__fadeIn"
          style="display: none;"
        >
          No hero with 
          <b />
        </div>
      </div>
    </div>
  </div>
</div>
`;
